core/src/io/anuke/mindustry/editor/MapRenderer.java

```diff
@@ -97,59 +101,91 @@ public void updateAll(){
     }
 
     private void render(int wx, int wy){
-        int x = wx / chunkSize, y = wy / chunkSize;
-        IndexedRenderer mesh = chunks[x][y];
-        Tile tile = editor.tiles()[wx][wy];
+        final Point windowPoint = new Point(wx, wy);
+        final Point gridPoint = new Point(wx/chunkSize, wy/chunkSize);
 
-        Team team = tile.getTeam();
-        Block floor = tile.floor();
+        final Point chunkStartPoint = getChunkStartPoint(windowPoint);
+        final int chunkArea = chunkSize * chunkSize;
+
+
+        IndexedRenderer mesh = getMesh(gridPoint);
+        Tile tile = getTile(windowPoint);
         Block wall = tile.block();
 
-        TextureRegion region;
+        int indexWall = chunkStartPoint.x + chunkStartPoint.y;
+        int indexDecal = chunkStartPoint.x  + chunkStartPoint.y + chunkArea;
 
-        int idxWall = (wx % chunkSize) + (wy % chunkSize) * chunkSize;
-        int idxDecal = (wx % chunkSize) + (wy % chunkSize) * chunkSize + chunkSize * chunkSize;
+        renderMeshRegion(windowPoint, mesh, tile, wall, indexWall);
 
-        if(wall != Blocks.air && (wall.synthetic() || wall instanceof BlockPart)){
-            region = !Core.atlas.isFound(wall.editorIcon()) ? Core.atlas.find("clear-editor") : wall.editorIcon();
 
-            if(wall.rotate){
-                mesh.draw(idxWall, region,
-                wx * tilesize + wall.offset(), wy * tilesize + wall.offset(),
-                region.getWidth() * Draw.scl, region.getHeight() * Draw.scl, tile.rotation() * 90 - 90);
-            }else{
-                mesh.draw(idxWall, region,
-                wx * tilesize + wall.offset() + (tilesize - region.getWidth() * Draw.scl) / 2f,
-                wy * tilesize + wall.offset() + (tilesize - region.getHeight() * Draw.scl) / 2f,
-                region.getWidth() * Draw.scl, region.getHeight() * Draw.scl);
-            }
-        }else{
-            region = floor.editorVariantRegions()[Mathf.randomSeed(idxWall, 0, floor.editorVariantRegions().length - 1)];
+        renderTileRegion(windowPoint, mesh, tile, wall, indexWall, indexDecal);
+    }
 
-            mesh.draw(idxWall, region, wx * tilesize, wy * tilesize, 8, 8);
-        }
+    private void renderTileRegion(Point windowPoint, IndexedRenderer mesh, Tile tile, Block wall, int indexWall, int indexDecal) {
+        TextureRegion textureRegion;
+        final boolean isEditorIconExist = Core.atlas.isFound(wall.editorIcon());
 
-        float offsetX = -(wall.size / 3) * tilesize, offsetY = -(wall.size / 3) * tilesize;
+        Float tileRegion = new Float();
```

Comment:
This is extremely confusing instantiation syntax. It looks exactly like you're creating a new wrapper `Float`, and not a `Rectangle2D.Float`. Only after looking at the imports and usage can you figure out what's going on.

Reply:
Originally, I used the `Rectangle2D.Float something = Rectangle2D.Float(x, y, w, h)` like that. But I ran my code formatting program and this was happened :( I'm really sorry about this to make you confusing.